dashboard/app/controllers/registrations_controller.rb

```diff
@@ -51,6 +55,29 @@ def create
     end
   end
 
+  def upgrade
+    return head(:bad_request) if params[:user].nil?
+    @user = User.find(current_user.id)
```

Comment:
As stated above, this is relatively meaningless. The current_user is the logged-in user, or nil if no user is logged in. There is no reason to find the user again, either way you do it. If we really want `@user` for some reason, simply assign the value of `current_user` to it. If we want to ensure the user is logged in, check `current_user.nil?`  or `user_signed_in?`. See http://devise.plataformatec.com.br/#controller-filters-and-helpers